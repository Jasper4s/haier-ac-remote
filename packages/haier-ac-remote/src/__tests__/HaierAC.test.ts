import { HaierAC } from '../HaierAC';
import * as rawCommands from '../lib/raw-commands';
import { toHex } from '../lib/toHex';

describe('HaierAC', () => {
  let ac: HaierAC;

  describe('state$', () => {
    beforeEach(() => {
      ac = new HaierAC({ ip: '0.0.0.0', mac: '' });
    });

    it('should work', () => {
      // @ts-ignore
      ac._rawData$.next(
        toHex(
          `00 00 27 15 00 00 00 00`,
          rawCommands.zero16(),
          rawCommands.zero16(),
          rawCommands.macAddress('00:07:A8:17:E9:AC'),
          rawCommands.zero16(),
          `00 00 00 01 00 00 00 25`,
          `ff ff 22 00 00 00 00 00`,
          `01 06 6d 01 00 15 00 00`,
          `00 7f 00 00 00 00 00 02`,
          `00 02 00 00 00 09 00 01`,
          `00 00 00 0c 45`,
        ),
      );

      expect(ac.state$.value).toMatchSnapshot();
    });
  });
});
