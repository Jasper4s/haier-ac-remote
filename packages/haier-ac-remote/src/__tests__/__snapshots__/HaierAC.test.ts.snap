// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HaierAC state$ should work 1`] = `
Object {
  "currentTemperature": 21,
  "fanSpeed": 2,
  "health": 1,
  "limits": 0,
  "mode": 2,
  "power": 9,
  "targetTemperature": 28,
}
`;
